lightgbm/src/test/scala/com/microsoft/azure/synapse/ml/lightgbm/split1/VerifyLightGBMCommon.scala

```diff
@@ -74,4 +92,206 @@ class VerifyLightGBMCommon extends TestBase {
+      println(s"** Start Measurement $i")
+      val _ = measuredModel.fit(train)
+      measures(i) = measuredModel.getPerformanceMeasures.get
+      println(s"Total time, ${measures(i).totalTime}")
+      println(s"Column statistics, ${measures(i).columnStatisticsTime()}")
+      println(s"Row statistics time, ${measures(i).rowStatisticsTime()}")
+      println(s"Row count time, ${measures(i).rowCountTime()}")
+      println(s"Sampling time, ${measures(i).samplingTime()}")
+      println(s"Training time, ${measures(i).trainingTime()}")
+      println(s"Overhead time, ${measures(i).overheadTime}")
+      println(s"Task total times, ${measures(i).taskTotalTimes().mkString(",")}")
+      println(s"Task overhead times, ${measures(i).taskOverheadTimes().mkString(",")}")
+      println(s"Task initialization times, ${measures(i).taskInitializationTimes().mkString(",")}")
+      println(s"Task library initialization times, ${measures(i).taskLibraryInitializationTimes().mkString(",")}")
+      println(s"Task network initialization times, ${measures(i).taskNetworkInitializationTimes().mkString(",")}")
+      println(s"Task data preparation times, ${measures(i).taskDataPreparationTimes().mkString(",")}")
+      println(s"Task dataset wait times, ${measures(i).taskWaitTimes().mkString(",")}")
+      println(s"Task dataset creation times, ${measures(i).taskDatasetCreationTimes().mkString(",")}")
+      println(s"Task training iteration times, ${measures(i).taskTrainingIterationTimes().mkString(",")}")
+      println(s"Task cleanup times, ${measures(i).taskCleanupTimes().mkString(",")}")
+      println(s"** Completed Measurement $i")
+    })
+    println(s"***** Averaged results for $measurementCount runs")
+    printMedianMeasure("Median Total time", measures, m => m.totalTime)
+    printMedianMeasure("Median Column statistics", measures, m => m.columnStatisticsTime())
+    printMedianMeasure("Median Row count time", measures, m => m.rowCountTime())
+    printMedianMeasure("Median Sampling time", measures, m => m.samplingTime())
+    printMedianMeasure("Median Row statistics time", measures, m => m.rowStatisticsTime())
+    printMedianMeasure("Median Training time", measures, m => m.trainingTime())
+    printMedianMeasure("Median Overhead time", measures, m => m.overheadTime)
+    printMedianMeasure("Median-max Task total times", measures, m => m.taskTotalTimes().max)
+    printMedianMeasure("Median-max Task overhead times", measures, m => m.taskOverheadTimes().max)
+    printMedianMeasure("Median-max Task initialization times", measures, m => m.taskInitializationTimes().max)
+    printMedianMeasure("Median-max Task data preparation times", measures, m => m.taskDataPreparationTimes().max)
+    printMedianMeasure("Median-max Task dataset creation times", measures, m => m.taskDatasetCreationTimes().max)
+    printMedianMeasure("Median-max Task training iteration times", measures, m => m.taskTrainingIterationTimes().max)
+  }
```

Comment:
this perf metrics code looks a little repetitive, perhaps it could be refactored to do this without a types API so you can just loop over metrics and print out these different variants

Reply:
This is really just a temporary util that I made an ignored test for now, which is why I didn't try and optimize it. It was some quick and dirty code. :) It was for some basic benchmarking, and I will refactor it later as part of doing more official perf analysis/benchmarking on the streaming stuff. I'll either make it a real util somewhere or refactor it as you said to be less repetitive.

Reply:
in some future PR  :)